lib/qa/authorities/linked_data/rdf_helper.rb

```diff
@@ -0,0 +1,102 @@
+require 'rdf'
+
+# Encapsulates processing of RDF results returned by the linked data authority.  This is used exclussively by Qa::Authorities::LinkedData::GenericAuthority
+# @see Qa::Authorities::LinkedData::GenericAuthority
+module Qa::Authorities
+  module LinkedData
+    module RdfHelper
+      private
+
+        def object_value(stmt_hash, consolidated_hash, name, as_string = true)
+          new_object_value = stmt_hash[name]
+          new_object_value = new_object_value.to_s if as_string
+          all_object_values = consolidated_hash[name] || []
+          all_object_values << new_object_value unless new_object_value.nil? || all_object_values.include?(new_object_value)
+          all_object_values
+        end
+
+        def init_consolidated_hash(consolidated_results, uri, id)
+          consolidated_hash = consolidated_results[uri] || {}
+          if consolidated_hash.empty?
+            consolidated_hash[:id] = uri
+            consolidated_hash[:id] = id unless id.nil? || id.length <= 0
+          end
+          consolidated_hash
+        end
+
+        def get_linked_data(url)
+          begin
+            graph = RDF::Graph.load(url)
+          rescue IOError => e
+            process_error(e, url)
+          end
+          graph
+        end
+
+        def process_error(e, url)
+          uri = URI(url)
+          response_code = ioerror_code(e)
+          case response_code
+          when 'format'
+            raise RDF::FormatError, "Unknown RDF format of results returned by #{uri}. (RDF::FormatError)  You may need to include gem 'linkeddata'."
+          when '404'
+            raise Qa::TermNotFound, "#{uri} Not Found - Term may not exist at LOD Authority. (HTTPNotFound - 404)"
+          when '500'
+            raise Qa::ServiceUnavailable, "#{uri.hostname} on port #{uri.port} is not responding.  Try again later. (HTTPServerError - 500)"
+          when '503'
+            raise Qa::ServiceUnavailable, "#{uri.hostname} on port #{uri.port} is not responding.  Try again later. (HTTPServiceUnavailable - 503)"
+          else
+            raise Qa::ServiceUnavailable, "Unknown error for #{uri.hostname} on port #{uri.port}.  Try again later. (Cause - #{e.message})"
+          end
```

Comment:
I don't believe QA handles timeouts for any authorities, linked data or otherwise.  It might be nice to add that kind of configuration to QA, but it should probably QA wide and not linked data specific.  An issue can be added for this after the PR is merged.